### PR TITLE
Search and Filter features 

### PR DESCRIPTION
**Pull request by iSayZ**

### CODE REVIEW

**Review by EmilieLgt — APPROVED** [review]

File: client/src/components/Shop/Card/FilterButtonContainer/FilterButtonByOrder/FilterButtonByOrder.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from "prop-types";
+import { useNavigate, useLocation } from 'react-router-dom';
+
+function FilterButtonByOrder ({ dataCategory }) {
+
+    const location = useLocation();
+    const searchParams = new URLSearchParams(location.search);
+    const query = searchParams.get("query");
+
+    const queryContain = query && query.includes('all');
```
Comment (fantinerudent) [inline client/src/components/Shop/Card/FilterButtonContainer/FilterButtonByOrder/FilterButtonByOrder.jsx]: ce fichier mériterait des commentaires aussi :)

File: client/src/components/Shop/Card/FilterButtonContainer/FilterMenu/FilterMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+function FilterMenu({ visible, setVisible, dataCategory }) {
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const query = searchParams.get("query");
+
+  const navigate = useNavigate();
+
+  // Si l'URL contient all (filtre pour rechercher dans toutes les catégories) l'état est true sinon false.
+  const [allCategory, setAllCategory] = useState(false);
```
Comment (fantinerudent) [inline client/src/components/Shop/Card/FilterButtonContainer/FilterMenu/FilterMenu.jsx]: Categories, allCategories

**Review by fantinerudent — CHANGES_REQUESTED** [review]

File: client/src/components/Shop/Card/FilterButtonContainer/FilterButtonByOrder/FilterButtonByOrder.jsx (hunk 1 above)
Comment (iSayZ) [reply client/src/components/Shop/Card/FilterButtonContainer/FilterButtonByOrder/FilterButtonByOrder.jsx]: I have added a comment into this component :)

File: client/src/components/Shop/Card/FilterButtonContainer/FilterMenu/FilterMenu.jsx (hunk 2 above)
Comment (iSayZ) [reply client/src/components/Shop/Card/FilterButtonContainer/FilterMenu/FilterMenu.jsx]: It's Ok, I have renamed my state

**Review by fantinerudent — APPROVED** [review]

**Pull request merged**